services/gitlab/gitlab-coverage.service.js

```diff
@@ -0,0 +1,121 @@
+'use strict'
+
+const Joi = require('@hapi/joi')
+const {
+  coveragePercentage: coveragePercentageColor,
+} = require('../color-formatters')
+const { optionalUrl } = require('../validators')
+const { BaseSvgScrapingService, NotFound, redirector } = require('..')
+
+const badgeSchema = Joi.object({
+  message: Joi.alternatives()
+    .try(
+      Joi.string().regex(/^([0-9]+\.[0-9]+%)|unknown$/),
+      Joi.equal('unknown')
```

Comment:
The catch here is that GitLab just RegEx's the coverage from the log of the executed commands, so ideally this should cover most cases IMO

I think unknown was added when the badge didn't work for the non-existent branch/coverage not set up situation

Reply:
> I think unknown was added when the badge didn't work for the non-existent branch/coverage not set up situation

There shouldn't be a need to have multiple paths to catch `unknown` though. Either the alternative can be dropped since the regex will allow `unknown` or the regex could be simplified to drop `unknown` since there's an alternative schema option to handle

Reply:
Removed line 14